GVRf/Framework/jni/objects/components/camera_rig_jni.cpp

```diff
@@ -197,16 +195,22 @@ Java_org_gearvrf_NativeCameraRig_setCameraSeparationDistance(
     camera_rig->set_camera_separation_distance(distance);
 }
 
-JNIEXPORT jfloat JNICALL
+JNIEXPORT jfloatArray JNICALL
 Java_org_gearvrf_NativeCameraRig_getFloat(JNIEnv * env,
         jobject obj, jlong jcamera_rig, jstring key) {
     std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
             CameraRig>*>(jcamera_rig);
     const char* char_key = env->GetStringUTFChars(key, 0);
     std::string native_key = std::string(char_key);
-    jfloat camera_rig_float = camera_rig->getFloat(native_key);
     env->ReleaseStringUTFChars(key, char_key);
-    return camera_rig_float;
+    try {
+        jfloat value = camera_rig->getFloat(native_key);
+        jfloatArray result_float = env->NewFloatArray(1);
+        env->SetFloatArrayRegion(result_float, 0, 1, &value);
+        return result_float;
+    } catch (const std::string & err) {
+        return NULL;
```

Comment:
> As you mentioned mixing up error codes (object not found) with data value (NaN or whatever) is not a good idea. We can easily think of many many reasons why it is bad. We have to avoid faking error codes wherever possible. I believe using NULL is one of most natural ways in handling object-not-found, either it is vector or float, and consistent across our cods too. Performance difference on this must be trivial.

Sorry, @jason2kim , I'm with Chao on this one: Creating an array to return a single `float` is adding a lot of overhead to the normal case. It would be reasonable if every possible `float` were a valid result, but that's not the case here. A `NaN` is "not a number" - and can't be passed in - so it does make sense to use it as an error code.

What you have coded is akin to a `find()` method returning an `Integer` so it can return `null` on not-found, instead of simply returning a `-1`. You wouldn't do that, because of the overhead of creating the reference type and then the cost of garbage collecting it ... and the same reasoning applies here.